src/Application/Application.php

```diff
@@ -172,7 +172,7 @@ public function processException($e)
 			$this->httpResponse->setCode($e instanceof BadRequestException ? ($e->getCode() ?: 404) : 500);
 		}
 
-		$args = ['exception' => $e, 'request' => end($this->requests) ?: NULL];
+		$args = ['exception' => $e, 'request' => end($this->requests) ?: $e->getRequest()];
```

Comment:
Hm, yes. If Router::match() fails, it can make problems. That check of type of $e is needed.